echopype/echodata/utils_platform.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+
+
+def _extvar_properties(ds, name):
+    """Test the external variable for presence and all-nan values,
+    and extract its time dimension name.
+    Returns <presence>, <valid values>, <time dim name>
+    """
+    if name in ds:
+        # Assumes the only dimension in the variable is a time dimension
+        time_dim_name = ds[name].dims[0] if len(ds[name].dims) > 0 else "scalar"
+        if not ds[name].isnull().all():
+            return True, True, time_dim_name
+        else:
+            return True, False, time_dim_name
+    else:
+        return False, False, None
+
+
+# clip incoming time to 1 less than min of EchoData["Sonar/Beam_group1"]["ping_time"] and
+#   1 greater than max of EchoData["Sonar/Beam_group1"]["ping_time"]
+# account for unsorted external time by checking whether each time value is between
+#   min and max ping_time instead of finding the 2 external times corresponding to the
+#   min and max ping_time and taking all the times between those indices
```

Comment:
Could you please put these into proper docstring under `_clip_by_time_dim`?

Reply:
Updated!